File: sdk-packaging/examples/rust/basic_chat.rs
// nxuskit SDK Example — Basic Chat (Rust)
//
// Uses the bundled nxuskit crate as a path dependency.
//
// Setup:
//   export NXUSKIT_SDK_DIR="/path/to/nxuskit-sdk-<version>-<platform>"
//   export OPENAI_API_KEY="sk-..."
//
// Run:
//   cd examples/rust
//   cargo run

use nxuskit::{ChatRequest, Message, NxuskitProvider, ProviderConfig};

fn main() -> Result<(), nxuskit::NxuskitError> {
    // Create a provider — reads OPENAI_API_KEY from environment
    let provider = NxuskitProvider::new(ProviderConfig {
        provider_type: "openai".into(),
        ..Default::default()
    })?;

    // Build a chat request using the builder pattern
    let request = ChatRequest::new("gpt-4o-mini")
        .with_message(Message::user("What is the capital of France? Reply in one sentence."))
        .with_max_tokens(100);

    // Send the request
    let response = provider.chat(request)?;

    println!("Response: {}", response.content);

    println!(
        "Tokens: {} prompt + {} completion",
        response.usage.estimated.prompt_tokens,
        response.usage.estimated.completion_tokens,
    );

    Ok(())
}

File: sdk-packaging/examples/python/basic_chat.py
"""nxuskit SDK Example — Basic Chat (Python)

Demonstrates: provider creation, synchronous chat, response reading, cleanup.

Run:
    export OPENAI_API_KEY="sk-..."
    python basic_chat.py

Prerequisites:
    export NXUSKIT_SDK_DIR="/path/to/nxuskit-sdk-1.0.4-oss-<platform>"
    export PYTHONPATH="$NXUSKIT_SDK_DIR/python/src:${PYTHONPATH:-}"
"""

import os
import sys

from nxuskit._ffi_provider import create_ffi_provider
from nxuskit._ffi_errors import ConfigError, ProviderError

def main():
    api_key = os.environ.get("OPENAI_API_KEY")
    if not api_key:
        print("Error: set OPENAI_API_KEY environment variable", file=sys.stderr)
        sys.exit(1)

    # Create an OpenAI provider
    with create_ffi_provider({
        "provider_type": "openai",
        "api_key": api_key,
    }) as provider:
        # Send a chat request
        response = provider.chat({
            "model": "gpt-4o-mini",
            "messages": [
                {"role": "user", "content": "What is the capital of France? Reply in one sentence."}
            ],
            "max_tokens": 100,
        })

        print(f"Response: {response.content}")
        print(f"Model: {response.model}")

        if response.usage:
            print(
                f"Tokens: {response.usage.prompt_tokens} prompt "
                f"+ {response.usage.completion_tokens} completion "
                f"= {response.usage.total_tokens} total"
            )


if __name__ == "__main__":
    try:
        main()
    except ConfigError as e:
        print(f"Configuration error: {e.message}", file=sys.stderr)
        sys.exit(1)
    except ProviderError as e:
        print(f"Provider error: {e.message}", file=sys.stderr)
        sys.exit(1)
